app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt

```diff
@@ -68,6 +68,22 @@ constructor(private val db: FirebaseFirestore, private val auth: FirebaseAuth) :
         .addOnFailureListener(onFailure)
   }
 
+  override fun userExists(
+      user: User,
+      onSuccess: (Boolean) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .whereEqualTo("userId", user.public.userId)
+        .whereEqualTo("username", user.public.username)
```

Comment:
Yes that's correct thanks a lot! The `SignInScreen` passes the temporary `User` that only have the default info given by Firestore. The `username` check might be removed since it involves useless loads from and to Firestore.